lib/src/txn.rs

```diff
@@ -102,20 +141,23 @@ impl Txn {
     }
 
     /// Commits the transaction in progress
-    pub async fn commit(mut self) -> Result<()> {
+    pub async fn commit(mut self) -> Result<Option<String>> {
         #[cfg(not(feature = "unstable-bolt-protocol-impl-v2"))]
```

Comment:
Can we move the entire function behind the feature flag?
Without the feature, it should still return `Result<()>`.
Essentially, have it in both versions and move the feature check just above the function and its docstring